src/engine/ui/widgets/Slider.cxx

```diff
@@ -81,8 +84,18 @@ void Slider::onMouseMove(const SDL_Event &event)
     {
       sliderButton.x = event.motion.x - sliderButton.w / 2;
       curVal = getValue(event.motion.x);
+      sliderSignal.emit(this);
```

Comment:
Lines 85-87 and Lines 59-61: code duplication.
Extract function.

Reply:
Good point. I was worried about what if in the future I want to extract the value from the button's position (i need to rename that function anyway) but I suppose we can just cross that bridge when we come to it.

Reply:
Actually, here I refer to 3 lines, not only you change. Lines 85,86,87 and 59,60,61

Reply:
I extracted the signal emission, but you're right, the other two lines are the same as well. I need to make some changes to the slider class, but I will handle that in a separate PR just for that.